alloy/src/main/java/com/liferay/faces/alloy/component/field/internal/FieldRenderer.java

```diff
@@ -60,12 +61,19 @@ private void encodeLabel(FacesContext facesContext, Field field, UIComponent lab
 
 		if (labelFacet != null) {
 
-			UIComponent checkboxLabelFacetChild = getSelectBooleanCheckboxLabelFacetChild(labelFacet);
+			SelectBooleanCheckbox checkboxLabelFacetChild = getSelectBooleanCheckboxLabelFacetChild(labelFacet);
 
 			if (checkboxLabelFacetChild != null) {
 
-				String checkboxClientId = checkboxLabelFacetChild.getClientId(facesContext);
-				responseWriter.writeAttribute("for", checkboxClientId, null);
+				String checkboxId = checkboxLabelFacetChild.getId();
+				boolean checkboxHasGeneratedId = (checkboxId == null || checkboxId.startsWith(UIViewRoot.UNIQUE_ID_PREFIX));
+				boolean checkboxHasClientBehaviors = !checkboxLabelFacetChild.getClientBehaviors().isEmpty();
+				boolean checkboxIdRendered = !checkboxHasGeneratedId || checkboxHasClientBehaviors;
+
+				if (checkboxIdRendered) {
+					String checkboxClientId = checkboxLabelFacetChild.getClientId(facesContext);
+					responseWriter.writeAttribute("for", checkboxClientId, null);
+				}
```

Comment:
@ngriffin7a, instead of checking if the checkbox id is rendered, we could always render ids on `alloy:selectBooleanCheckbox`. I'm not convinced that either solution is better, but let me know if you think that always rendering the id on `alloy:selectBooleanCheckbox` is better.